jamie-sgro/url-shortener
# Feature/#7 add db accessor class

## Proposed Changes

closes #7 

  - added a db_accessor class
  - added tests and interfaces for the same
  - added a factory for passing concrete implementations of the db_accessor around the system

## Suggested Review Steps

 - ensure tests greenlight
 - start from the db_accessor, then f12 in vscode to review all references

## Checklist

- [x] I've added documentation where possible
- [x] I've added unit tests where possible
